src/goLanguageServer.ts

```diff
@@ -97,28 +133,42 @@ async function startLanguageServer(ctx: vscode.ExtensionContext, config: Languag
 	// Check if we should recreate the language client. This may be necessary
 	// if the user has changed settings in their config.
 	if (!deepEqual(latestConfig, config)) {
-		// Track the latest config used to start the language server.
+		// Track the latest config used to start the language server,
+		// and rebuild the language client.
 		latestConfig = config;
+		languageClient = await buildLanguageClient(config);
```

Comment:
(optional) wondering if it makes more sense to update the latestConfig with the new value inside buildLanguageClient because basically latestConfig is the config used to build the current languageClient, right? And, if the languageClient becomes null, I think latestConfig need to be null.

Reply:
I'll change it so that we don't set it to null if it's disabled. That way it will still work fine, and we won't have to change lots of variables around.